tests/load_data/event_data/test_opta.py

```diff
@@ -0,0 +1,190 @@
+                "start_y": [5.0, -5.0, 5.0, 0.01, -0.93, -0.57, 5.0, -2.68, -2.68],
+                "datetime": np.array(
+                    [
+                        "2023-01-22T10:28:32.117",
+                        "2023-01-22T11:18:32.152",
+                        "2023-01-22T11:18:32.152",
+                        "2023-01-22T11:18:33.637",
+                        "2023-01-22T11:18:36.207",
+                        "2023-01-22T11:18:39.109",
+                        "2023-01-22T11:18:41.615",
+                        "2023-01-22T11:18:43.119",
+                        "2023-01-22T11:18:43.120",
+                    ],
+                    dtype="datetime64",
+                ),
+                "player_name": [
+                    np.nan,
+                    np.nan,
+                    np.nan,
+                    "Anas Tahiri",
+                    "Timo Zaal",
+                    "Timo Zaal",
+                    "Johan Hove",
+                    "Timo Zaal",
+                    "Johan Hove",
+                ],
+            }
+        )
+
+    def test_load_opta_event_data(self):
+
+        metadata, event_data = load_opta_event_data(
+            self.f7_loc, self.f24_loc, pitch_dimensions=[10, 10]
+        )
+        pd.testing.assert_frame_equal(event_data, self.expected_event_data)
+        assert metadata.match_id == self.expected_metadata.match_id
+
+    def test_load_metadata(self):
+
+        metadata = _load_metadata(self.f7_loc)
+        assert metadata == self.expected_metadata
+
+    def test_get_player_info(self):
+        player_data = [
+            {
+                "PlayerRef": "s123",
+                "Formation_Place": "0",
+                "Position": "Substitute",
+                "SubPosition": "Midfielder",
+                "Status": "Substitute",
+                "ShirtNumber": "33",
+            },
+            {
+                "PlayerRef": "s234",
+                "Formation_Place": "1",
+                "Position": "GoalKeeper",
+                "Status": "Start",
+                "ShirtNumber": "2",
+            },
+        ]
+        player_names = {"123": "Sven Kerhoffs", "234": "Niels Smits"}
+
+        expected_result = pd.DataFrame(
+            {
+                "player_id": [123, 234],
+                "player_name": ["Sven Kerhoffs", "Niels Smits"],
+                "formation_place": [0, 1],
+                "position": ["midfielder", "goalkeeper"],
+                "starter": [False, True],
+                "shirt_number": [33, 2],
+            }
+        )
+
+        result = _get_player_info(player_data, player_names)
+        pd.testing.assert_frame_equal(result, expected_result)
+
+    def test_load_event_data(self):
+        event_data = _load_event_data(self.f24_loc)
+
+        # player name is added in other function later in the pipeline
+        expected_event_data = self.expected_event_data.copy().drop(
```

Comment:
zelfde als hierboven

Reply:
Is an arbitrary choice, but the metadata I needed twice, exactly the same. So the alternative is to write out the metadata class twice, I think this is more clear. And for the event data I only needed to make some small addjustments to make it fit the second time. I think this is more readable than adding duplicate code, do you agree?

Reply:
Yes